src/components/pages/NotFound/NotFound.test.tsx

```diff
@@ -0,0 +1,32 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import NotFound from './NotFound';
+import { BrowserRouter } from 'react-router-dom';
+import { describe, it, expect, vi } from 'vitest';
+
+const MockNotFound = () => {
+  return (
+    <BrowserRouter>
```

Comment:
You don't need ```<BrowserRouter>```here. Just mock useNavigate from 'react-router-dom' and then you can test mocked navigate - click the button "Go back" and assert mockNavigate to has been called with -1
```
import { fireEvent, render, screen } from '@testing-library/react';
import NotFound from './NotFound';
import { describe, it, expect, vi } from 'vitest';

const mockNavigate = vi.fn();

vi.mock('react-router-dom', () => ({
  useNavigate: vi.fn(() => mockNavigate),
}));

describe('404 Page component', () => {
...
```

Reply:
fixed, thank you, I didn't know how to do it